patterns/1-initial/scoring-grid.md

```diff
@@ -0,0 +1,114 @@
+## Title
+
+Software Project Scoring Grid
```

Comment:
Rob also mentioned "scorecard" elsewhere. Does scorecard have a specific meaning in the industry? I feel like I have heart it a couple of times already, but might be wrong. 

So an other alternative could be:
> Repository Scorecard

Reply:
> Added a couple of specific in-line comments but also have some general feedback: I'm unclear whether this is a pattern to help improve InnerSource repository setup (i.e. help contributors make better quality contributions), or whether it's a pattern to improve any and all aspects of engineering. I think both are valid things, but an InnerSource setup focus is probably more relevant specifically here?

> I think it would benefit from a more specific focus on InnerSource setup in context/forces sections for example, where I see the example scoring grid under solutions already being InnerSource orientated.

So evidentially my scorecard comes from a place of Code Quality and Engineering Leadership - the scorecard concept was inspired by work done at a previous employer to roll out InnerSource at the team and repository level. From a Code Quality perspective, the overlap between InnerSource best practice and Community best practice (working with a team) are quite strong. I think it would make sense to rewrite the example scorecard to highlight InnerSource best practice, e.g. from the *InnerSource Maturity Model* pattern - I can give that a go since it would be more appropriate to the DNA of the idea.
 
> We use this pattern in the 'InnerSource setup' context, and there are plenty of examples in our wider org in the more general engineering context (e.g. security scorecards, operational scorecards, engineering or test maturity assessments, delivery health dashboards).

Good point; I think it's worth calling out that the Scorecard pattern can be used to measure other things, as exampled by your list, and that the general pattern can be expanded if you want to promote other areas of best practice.

Reply:
I think the word **repository** should be used consistently in the pattern - I was struggling between project/service/codebase - but repository ("git repo") seems like the right focal point.

As for title variations:
- Repository Scoring Grid
- Repository Score Grid
- Repository Scorecard

A similar pattern common in the industry would be a Career Path Matrix or Job Levelling Matrix e.g. https://lattice.com/library/what-is-a-job-leveling-matrix

So an alternative name could be:
- Repository Maturity Matrix

I think a scorecard is an artefact of scoring, based on the grid - so I think my original verb "Scoring" is correct because the grid is used for scoring, i.e. producing scorecards - so I think my title preference is "Repository Scoring Grid" - and the example should be refactored to focus on InnerSource best practice at the repository level based on practical things at the repository level that intersect with Maturity Model.

Reply:
Naming is hard :)

Adding to the complexity that we already have a pattern called [Maturity Model](https://patterns.innersourcecommons.org/p/maturity-model), which we would not want to get confused with this new pattern.

Reply:
A thought -- how about "Repository Scoring" ..? 

It occurs to me the underlying pattern here is to guide the key stakeholders of a _repository_ to improve in a structured measureable way by _scoring_, which also allows a healthy dose of reporting/gamification activity when there exists a larger portfolio. 

The fact you use a 'scoring grid' with dimensions and grades is one way of doing the scoring. There are others e.g. a scorecard I'd consider to be slightly different in that it's primary purpose is to have a single overall grade or score that is easy to communicate, and the 'card' bit means there is additional detail that tells you why so you can improve. Simple 'number of failed checks' is another common approach (e.g. linter type semantics). 

So the question is really whether the pattern is the scoring _grid_, or whether the pattern is the _scoring_. If the latter then allow the grid to be one of a few different examples rather than leading the pattern in the title.

Reply:
> when there exists a larger portfolio

On this topic, I wince a bit in this pattern as you reference "poor engineering leadership", perhaps because I have needed this pattern as an engineering leader **chuckle**. IMO this pattern becomes necessary (and it's use a signal of good engineering leadership) based **on the size of the portfolio**. As the number of repositories scales up... how do you govern any standards that you have? Well, this. 

So it might well be poor engineering leadership, but it's also often a successful org in a growth stage where the growing scale requires previously unwritten rules to be formalised.